## Recommended Secure coding - Specify an appropriate locale when comparing locale-dependent data

consider the  following java code snippet

```java

	public static void processTag(String tag) {
	  if (tag.toUpperCase().equals("SCRIPT")) {
		return;
	  }
	  // Process tag
	}


```

 This noncompliant code example uses the locale-dependent String.toUpperCase() method to convert an HTML tag to uppercase to check it for further processing. The code must ignore <SCRIPT> tags, as they indicate code that is to be discarded. Whereas the English locale would convert "script" to "SCRIPT", the Turkish locale will convert "script" to "SCRÄ°PT", and the check will fail to detect the <SCRIPT> tag.

 Rather better code is to specify the Locale, may be like this

 ```java
	 tag.toUpperCase(Locale.ENGLISH)
 ```

 Yet another apporach spefic to java, 
using equalIgnoreCase() compliant because compares two strings, one of which is plain ASCII, and therefore its behavior is well-understood. 

[source](https://wiki.sei.cmu.edu/confluence/display/java/STR02-J.+Specify+an+appropriate+locale+when+comparing+locale-dependent+data)
